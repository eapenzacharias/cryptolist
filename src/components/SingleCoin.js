/* eslint-disable no-unused-vars */
import PropTypes from 'prop-types';
import {
  Badge, Button, Col, Container, Modal, Row,
} from 'react-bootstrap';
import '../styles/model.scss';

const imgURL = 'https://cryptoicons.org/api/icon/';

const SingleCoin = (props) => {
  const {
    id, symbol, name,
    priceUsd, changePercent24Hr, supply,
    volumeUsd24Hr, marketCapUsd, onHide,
  } = props;

  return (
    <>
      <Modal
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...props}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
        className={`popup ${changePercent24Hr >= 0 ? 'rising' : 'falling'}`}
      >
        <Modal.Header
          closeButton
          className={`${changePercent24Hr >= 0 ? 'rising' : 'falling'}`}
        >
          <Container>
            <Row>
              <Col xs="3">
                <img src={`${imgURL}${symbol.toLowerCase()}/50`} alt={symbol} />
              </Col>
              <Col>
                <h2>{`${name} - ${symbol}`}</h2>
              </Col>
              <Col>
                <h3>{`$${priceUsd.toFixed(4)}`}</h3>
              </Col>
            </Row>
          </Container>
        </Modal.Header>
        <Modal.Body>
          <Container>
            <Row>
              <Col>
                <h3>Percentage Change 24H</h3>
                <h3 className={`${changePercent24Hr >= 0 ? 'headrising' : 'headfalling'}`}>
                  {`${changePercent24Hr.toFixed(4)}%`}
                </h3>
              </Col>
              <Col>
                <h3>Current Market Cap $</h3>
                <h3>
                  {`$${Math.floor(marketCapUsd)}`}
                </h3>
              </Col>
            </Row>
            <Row>
              <Col>
                <h3>Current Supply</h3>
                <h3>
                  {supply}
                </h3>
              </Col>
              <Col>
                <h3>Trade Volume 24H</h3>
                <h3>
                  {`$${volumeUsd24Hr.toFixed(0)}`}
                </h3>
              </Col>
            </Row>
          </Container>
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={onHide}>Close</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

SingleCoin.propTypes = {
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  supply: PropTypes.number.isRequired,
  changePercent24Hr: PropTypes.number.isRequired,
  priceUsd: PropTypes.number.isRequired,
  volumeUsd24Hr: PropTypes.number.isRequired,
  marketCapUsd: PropTypes.number.isRequired,
  onHide: PropTypes.func.isRequired,
};

export default SingleCoin;
