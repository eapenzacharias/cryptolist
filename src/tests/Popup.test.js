import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SingleCoin from '../components/SingleCoin';
import mockAPI from './__mock__/mockAPI';

describe('Should load the Homepage with menu', () => {
  const {
    id, symbol, name, priceUsd, changePercent24Hr, supply, volumeUsd24Hr, marketCapUsd,
  } = mockAPI[0];
  test('Snapshot matched', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <SingleCoin
          key={id}
          id={id}
          symbol={symbol}
          name={name}
          priceUsd={priceUsd}
          supply={supply}
          changePercent24Hr={changePercent24Hr}
          marketCapUsd={marketCapUsd}
          volumeUsd24Hr={volumeUsd24Hr}
          onHide={() => console.log('OnHide')}
        />
      </MemoryRouter>,
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
