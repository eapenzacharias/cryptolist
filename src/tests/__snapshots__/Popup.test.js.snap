// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should load popup od single Snapshot matched 1`] = `[Function]`;
